implementations/rust/ockam/ockam_command/src/markdown/mod.rs

```diff
@@ -256,31 +256,35 @@ fn generate_arg_markdown(buffer: &mut Vec<u8>, arg: &clap::Arg) -> io::Result<()
         Some([]) => unreachable!(),
         None => arg.get_id().to_string().to_ascii_uppercase(),
     };
+    let formatted_value_name = match arg.is_required_set() {
+        true => format!("<{value_name}>"),
+        false => format!("[{value_name}], (optional)"),
+    };
 
     match (arg.get_short(), arg.get_long()) {
         (Some(short), Some(long)) => {
             if arg.get_action().takes_values() {
-                write!(buffer, "`-{short}`, `--{long} <{value_name}>`")?
+                write!(buffer, "`-{short}`, `--{long} {formatted_value_name}`")?
```

Comment:
A small detail here: note that the `, (optional)` piece should be left out of the code block. Also, now that I take a second look at this, I think it feels better without the `,` before the `(optional)`. 

To avoid hacky stuff, maybe the cleanest approach would be to return two values so you can easily handle the code blocks:

```rust
let (value_name, optional) = match arg.is_required_set() {
    true => (format!("<{value_name}>"), ""),
    false => (format!("[{value_name}]"), "(optional)"),
};
```

```rust
write!(buffer, "`-{short}`, `--{long} {value_name}` {optional}")?
```

Reply:
Fixed!